modules/apps/object/object-service/src/main/java/com/liferay/object/internal/system/SystemObjectDefinitionMetadataRegister.java

```diff
@@ -0,0 +1,142 @@
+/**
```

Comment:
I think we have already a pattern for these type of classes.

Take a look at: https://github.com/liferay/liferay-portal/blob/b3173da81b62430f7150b695d5e944b224279580/modules/apps/export-import/export-import-resources-importer/src/main/java/com/liferay/exportimport/resources/importer/internal/extender/ResourceImporterExtender.java#L55

From what I see we can call ServiceTrackerFactory.create to create the tracker and declare an inner private class for the ServiceTrackerCustomizer.

I also see that there is a naming pattern to call these classes ...Extender, but I'm not sure if this applies to our use case. I bet that it could

Reply:
Perfect! I will change it to follow the pattern.